packages/schematics/angular/migrations/update-14/update-tsconfig-target.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule, Tree } from '@angular-devkit/schematics';
+import { JSONFile } from '../../utility/json-file';
+import { getWorkspace } from '../../utility/workspace';
+import { Builders } from '../../utility/workspace-models';
+
+const SUPPORTED_BUILDERS = [Builders.Karma, Builders.NgPackagr, Builders.Browser];
+
+/** Migration to update tsconfig compilation target option to es2020. */
+export default function (): Rule {
+  return async (host) => {
+    // Workspace level tsconfig
+    updateTargetToES2020(host, 'tsconfig.json');
+
+    // Find all tsconfig which are refereced by the builders
+    const workspace = await getWorkspace(host);
+    for (const [, project] of workspace.projects) {
+      for (const [, target] of project.targets) {
+        if (!SUPPORTED_BUILDERS.includes(target.builder as Builders)) {
+          // Unknown builder
+          continue;
+        }
+
+        // Update all other known CLI builders that use a tsconfig
+        const tsConfigs = [target.options || {}, ...Object.values(target.configurations || {})]
+          .filter((opt) => typeof opt?.tsConfig === 'string')
+          .map((opt) => (opt as { tsConfig: string }).tsConfig);
+
+        const uniqueTsConfigs = new Set(tsConfigs);
+        if (uniqueTsConfigs.size < 1) {
+          continue;
+        }
+
+        for (const tsConfig of uniqueTsConfigs) {
+          updateTargetToES2020(host, tsConfig);
+        }
+      }
+    }
+  };
+}
+
+function updateTargetToES2020(host: Tree, tsConfigPath: string) {
+  const json = new JSONFile(host, tsConfigPath);
+  const target = json.get(['compilerOptions', 'target']);
+
+  if (typeof target === 'string' && target.toLowerCase() !== 'es2020') {
```

Comment:
Should we only update if target < ES2020? If someone was running on ES2021 or next, we shouldn't downgrade them.

Reply:
Good call.